diff --git a/llvm/lib/IR/AsmWriter.cpp b/llvm/lib/IR/AsmWriter.cpp
index 5cf9532a988f..5b128263f8fe 100644
--- a/llvm/lib/IR/AsmWriter.cpp
+++ b/llvm/lib/IR/AsmWriter.cpp
@@ -3091,7 +3091,7 @@ void AssemblyWriter::printModule(const Module *M) {

   if (ShouldPreserveUseListOrder)
     UseListOrders = predictUseListOrder(M);
-
+/*
   if (!M->getModuleIdentifier().empty() &&
       // Don't print the ID if it will start a new line (which would
       // require a comment char before it).
@@ -3126,6 +3126,7 @@ void AssemblyWriter::printModule(const Module *M) {
       Out << "\"\n";
     } while (!Asm.empty());
   }
+  */

   printTypeIdentities();

@@ -4294,7 +4295,7 @@ void AssemblyWriter::printBasicBlock(const BasicBlock *BB) {
     else
       Out << "<badref>:";
   }
-
+#if false
   if (!IsEntryBlock) {
     // Output predecessors for the block.
     Out.PadToColumn(50);
@@ -4310,6 +4311,7 @@ void AssemblyWriter::printBasicBlock(const BasicBlock *BB) {
       }
     }
   }
+#endif

   Out << "\n";

